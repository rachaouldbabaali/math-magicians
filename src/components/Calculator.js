import React, { useState } from 'react';
import PropTypes from 'prop-types';
import calculate from '../logic/calculate';

function Calculator({ result, setResult }) {
  const [output, setOutput] = useState('0');

  const getButtonName = (e) => {
    const buttonName = e.target.textContent;
    setResult(calculate(result, buttonName));
  };

  React.useEffect(() => {
    setOutput(result.next || result.operation || result.total || '0');
  }, [result]);
  return (
    <div className="calcu">
      <h1 className="calcu-h1">Let&apos;s do some math!</h1>
      <div className="calculator-wrap">
        <div className="calculator-display">{output}</div>
        <div className="calculator-keypads">
          <button className="key1" type="button" onClick={getButtonName}>AC</button>
          <button className="key2" type="button" onClick={getButtonName}>+/-</button>
          <button className="key3" type="button" onClick={getButtonName}>%</button>
          <button className="key4" type="button" onClick={getButtonName}>&#247;</button>
          <button className="key5" type="button" onClick={getButtonName}>7</button>
          <button type="button" onClick={getButtonName}>8</button>
          <button type="button" onClick={getButtonName}>9</button>
          <button className="key8" type="button" onClick={getButtonName}>x</button>
          <button type="button" onClick={getButtonName}>4</button>
          <button type="button" onClick={getButtonName}>5</button>
          <button type="button" onClick={getButtonName}>6</button>
          <button className="key12" type="button" onClick={getButtonName}>-</button>
          <button type="button" onClick={getButtonName}>1</button>
          <button type="button" onClick={getButtonName}>2</button>
          <button type="button" onClick={getButtonName}>3</button>
          <button className="key12" type="button" onClick={getButtonName}>+</button>
          <button className="key17" type="button" onClick={getButtonName}>0</button>
          <button type="button" onClick={getButtonName}>.</button>
          <button className="key19" type="button" onClick={getButtonName}>=</button>
        </div>
      </div>
    </div>
  );
}

Calculator.propTypes = {
  result: PropTypes.shape({ total: null, next: null, operation: null }).isRequired,
  setResult: PropTypes.func.isRequired,
};

export default Calculator;
