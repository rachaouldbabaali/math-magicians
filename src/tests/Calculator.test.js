import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Calculator from '../components/Calculator';

describe('Calculator', () => {
  it('renders the component', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the component', () => {
    const { getByText } = render(
      <Calculator result={{}} setResult={() => {}} />,
    );
    expect(getByText("Let's do some math!")).toBeInTheDocument();
  });
});
