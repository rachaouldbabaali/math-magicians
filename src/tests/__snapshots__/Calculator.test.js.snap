// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator renders the component 1`] = `
<div
  className="calcu"
>
  <h1
    className="calcu-h1"
  >
    Let's do some math!
  </h1>
  <div
    className="calculator-wrap"
  >
    <div
      className="calculator-display"
    >
      0
    </div>
    <div
      className="calculator-keypads"
    >
      <button
        className="key1"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="key2"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="key3"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="key4"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="key5"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="key8"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="key12"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="key12"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="key17"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="key19"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
